src/test/java/com/taf/tests/steps/api/CommonApiDefinitionSteps.java

```diff
@@ -0,0 +1,15 @@
+package com.taf.tests.steps.api;
+
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.Given;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class CommonApiDefinitionSteps extends BasicApiDefinitionSteps {
+    @Given("User saves the following variables")
+    public void userSavesTheFollowingVariables(DataTable dataTable) {
+        Map<String, String> unmodifybleDataTableMap = dataTable.asMap(String.class, String.class);
```

Comment:
you can simplify it a little to `dataTable.asMap()` if you use String class for both key and value

Reply:
Fixed after code refactoring.